packages/example/src/pages/guides/v4-migration.mdx

```diff
@@ -0,0 +1,86 @@
+---
+title: V4.0 Migration
+description: Guide for migrating to v4.0 of the theme
+---
+
+<PageDescription>
+
+Carbon's Gatsby Theme uses version 5 of the Gatsby library. This update combines
+the control and scalability of server-side rendering with the performance of
+static-site generation. The major version change comes with the potential for
+breaking changes, this guide aims to cover the most likely issues a site built
+with the Carbon Gatsby theme might encounter. To have the smoothest possible
+experience, we recommend [upgrading to v3](/guides/v3-migration) of Carbon's
+Gatsby Theme first if you haven't already done so.
+
+</PageDescription>
+
+## Breaking changes
+
+1. Bumped minimum version of Gatsby to `v5`
+1. Gatsby files (i.e. gatsby-config.js and gatsby-node.js), need to be converted
+   to **.mjs** (node modules) files.
+1. TBD...
+
```

Comment:
Copying over some "breaking" things I noticed when migrating Carbon.

1. Markdown link with title needs double quotes
```jsx
// Before
[I'm a markdown link with title](https://www.carbondesignsystem.com "Carbon's Homepage")
// After
[I'm a markdown link with title](https://www.carbondesignsystem.com 'Carbon's Homepage')
```

1. HTML-style comments will no longer work inside of mdx files, you will need to use JavaScript-style comments with curly braces:
```jsx
// Before
<!-- Place comments here -->
// After
{/* Place comments here */}
```